File: isogram/Isogram.cs
using System;
using System.Linq;

public static class Isogram
{
    public static bool IsIsogram(string word) =>
        word.ToLowerInvariant()
            .Where(char.IsLetter)
			.GroupBy(letter => letter)
			.All(letter => letter.Count() == 1);
}

File: fsharp/isogram/Isogram.fs
module Isogram

open System

let onlyOne (_, values) =
    Seq.length values = 1

let isIsogram str =
    str
    |> Seq.filter Char.IsLetter
    |> Seq.map Char.ToLowerInvariant
    |> Seq.groupBy id
    |> Seq.forall onlyOne